File: Daily Test/Java/Bit Transfer - LSB To MSB 3.9.2025.java
import java.util.*;
class Main {
    static String toBinary(int x){
        return Integer.toBinaryString(x);
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] arr = new int[N];
        for(int i=0;i<N;i++) arr[i]=sc.nextInt();
        
        String[] bin = new String[N];
        for(int i=0;i<N;i++) bin[i] = toBinary(arr[i]);
        
        String extra = bin[N-1].charAt(bin[N-1].length()-1) + bin[0].substring(0, bin[0].length()-1);
        System.out.print(Integer.parseInt(extra,2) + " ");
        
        for(int i=1;i<N;i++){
            String tmp = bin[i-1].charAt(bin[i-1].length()-1) + bin[i].substring(0, bin[i].length()-1);
            System.out.print(Integer.parseInt(tmp,2) + " ");
        }
    }
}

File: Daily Test/C/Bit Transfer - LSB To MSB 3.9.2025.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main() {
    int N; scanf("%d", &N);
    int arr[N];
    for(int i=0; i<N; i++) scanf("%d", &arr[i]);
    
    char bin[N][50];
    for(int i=0; i<N; i++) {
        itoa(arr[i], bin[i], 2);
    }
    
    char extra[50];
    int len = strlen(bin[0]);
    sprintf(extra, "%c%.*s", bin[N-1][len-1], len-1, bin[0]);
    printf("%d ", (int)strtol(extra, NULL, 2));
    
    for(int i=1; i<N; i++) {
        len = strlen(bin[i]);
        char tmp[50];
        sprintf(tmp, "%c%.*s", bin[i-1][len-1], len-1, bin[i]);
        printf("%d ", (int)strtol(tmp, NULL, 2));
    }
    return 0;
}